gammapy/modeling/models/core.py

```diff
@@ -361,6 +362,46 @@ def to_dict(self):
         else:
             return {"components": models_data}
 
+    def to_parameters_table(self):
+        """Convert Models parameters to an astropy Table."""
+        pars = self.parameters.to_table()
+        rows = []
+        for name in self.parameters_unique_names:
+            row = {}
+            #Warning: splitting of parameters will break is source name has a "." in its name.
+            parts = name.split(".")
+            if len(parts) == 3:
+                row["model"], row["type"], _ = parts
+            elif len(parts) == 2:
+                row["model"], row["type"] = parts[0], ""
+            rows.append(row)
+        table = table_from_row_data(rows)
+        table = hstack([table, pars])
+        self._table_cached = table
+        return  table
+
+    def update_from_table(self, t):
```

Comment:
As you rely on the order of the parameters here, maybe we could just move this this to `Parameters.set_parameters_from_table()` or similar? Because we already have the API to update parameters like `Parameters.set_parameters_factors()`

Reply:
Changed name to `set_parameters_from_table` but still in `models/core.py`

Reply:
added tests. 
Passing on y local machine